app/models/manageiq/providers/azure/network_manager/floating_ip.rb

```diff
@@ -1,2 +1,5 @@
 class ManageIQ::Providers::Azure::NetworkManager::FloatingIp < ::FloatingIp
+  def self.available
+    joins(:network_port).where({ "network_ports.device_id" => nil }) + where(:network_port_id => nil)
+  end
```

Comment:
@djberg96 @blomquisg - expensive? I'm not sure how to optimize this given the associations that are in place.